From 4df08933bd5b527c5dd2befb2f36519b842c649d Mon Sep 17 00:00:00 2001
From: Ethan Chen <intervigil@gmail.com>
Date: Thu, 15 Oct 2015 13:50:10 -0700
Subject: [PATCH 01/45] Revert "egl: Remove old tuna BGRA vs RGBA workaround."

This reverts commit 733a80754786d39cdc0fee09509b194472c320bc.
and updated for Android O.

Change-Id: Ifb41720ed020489892a667914ea3bd3f1ac6504e
---
 opengl/libs/Android.bp     |  7 +++++++
 opengl/libs/EGL/eglApi.cpp | 11 +++++++++++
 2 files changed, 18 insertions(+)

diff --git a/opengl/libs/Android.bp b/opengl/libs/Android.bp
index 32c2d7e0d5..435d00cae8 100644
--- a/opengl/libs/Android.bp
+++ b/opengl/libs/Android.bp
@@ -124,6 +124,13 @@ cc_library_static {
 
 cc_library_shared {
     name: "libEGL",
+    product_variables: {
+        lineage: {
+            egl_workaround_bug_10194508: {
+                cppflags: ["-DWORKAROUND_BUG_10194508"],
+            },
+        },
+    },
     defaults: ["egl_libs_defaults"],
     srcs: [
         "EGL/egl_tls.cpp",
diff --git a/opengl/libs/EGL/eglApi.cpp b/opengl/libs/EGL/eglApi.cpp
index 94dfe6a9de..37891cbe07 100644
--- a/opengl/libs/EGL/eglApi.cpp
+++ b/opengl/libs/EGL/eglApi.cpp
@@ -615,6 +615,16 @@ void getNativePixelFormat(EGLDisplay dpy, egl_connection_t* cnx, EGLConfig confi
     EGLint componentType = EGL_COLOR_COMPONENT_TYPE_FIXED_EXT;
     cnx->egl.eglGetConfigAttrib(dpy, config, EGL_COLOR_COMPONENT_TYPE_EXT, &componentType);
 
+#if WORKAROUND_BUG_10194508
+    int intFormat = (int)(*format);
+    if (!cnx->egl.eglGetConfigAttrib(dpy, config, EGL_NATIVE_VISUAL_ID,
+            &intFormat)) {
+        ALOGE("eglGetConfigAttrib(EGL_NATIVE_VISUAL_ID) failed: %#x",
+                eglGetError());
+        format = 0;
+    }
+    *format = (android_pixel_format)intFormat;
+#else
     EGLint a = 0;
     EGLint r, g, b;
     r = g = b = 0;
@@ -662,6 +672,7 @@ void getNativePixelFormat(EGLDisplay dpy, egl_connection_t* cnx, EGLConfig confi
             format = HAL_PIXEL_FORMAT_RGBA_FP16;
         }
     }
+#endif
 }
 
 EGLSurface eglCreateWindowSurface(  EGLDisplay dpy, EGLConfig config,
-- 
2.17.0

